src/main/java/com/tananaev/jsonpatch/JsonPatch.java

```diff
@@ -26,7 +26,7 @@
 public class JsonPatch extends LinkedList<AbsOperation> {
 
     public JsonElement apply(JsonElement original) {
-        JsonElement result = original;
+        JsonElement result = original.deepCopy();
```

Comment:
Are you sure there are no regressions?

Reply:
Should not be. Doesn't deviate from the existing functionality where it returns a new patched object without modifying the input.